uPortal-core/src/main/java/org/apereo/portal/portlet/om/IPortletDefinition.java

```diff
@@ -59,8 +60,31 @@
      */
     boolean setPortletPreferences(List<IPortletPreference> portletPreferences);
 
+    /**
+     * The current lifecycle state is equal to the last entry in the lifecycle history that is not
+     * in the future.
+     */
     PortletLifecycleState getLifecycleState();
 
+    /**
+     * Adds the specified state to the lifecycle history at the present moment, removing any
+     * lifecycle changes scheduled for the future.
+     */
+    void setLifecycleState(PortletLifecycleState lifecycleState, IPerson user);
```

Comment:
This method is silently destructive. Better to return `boolean` indicating whether scheduled future lifecycle changes were removed?  Would a caller ever care? I guess could add the boolean in the future if eventually someone cares.